### DISCUSSION TITLE
Is there a way to use your scripts as a an importer to lidarr from other downloaders ?

### DISCUSSION THREAD

**Discussion by neilanthonystuckey** [RandomNinjaAtk/docker-lidarr-extended#151] [General]
I tend to use soulseek to find odd ball stuff that Lidarr wont grab or is hard to find.  Can one of the scripts in the /scripts folder be used to mass import from a folder ? I think it would be fairly easy to make a simple nzbget script to run on of your scripts after download is completed. My soulseek client has an option to run scripts after downloads as well.  

thanks for your time

Neil

**Comment by RandomNinjaAtk** [comment]
It would be pretty trivial to have a script to tell lidarr to scan/import a folder, thats essentially what this does now.... But lidarr expects folders to be named a certain way, so if your client cannot name them in a format lidarr will accept, that will be your issue.

**Reply by neilanthonystuckey** [reply]
Hey thanks for the fast reply.  So if i just dump audio folders/files into the lidarr completed downloads folder and use filebot to rename based on music brains id or match lidarr's current naming scheme then lidarr should import the audio files at some point ? Even if a client or app doesn't tell it a download is ready for import ? I tried a few different things but i can't seem to get lidarr to import from completed downloads folder unless it comes from your Audio.sh script or from nzbget. I know i could do this with Beets import but i tend to get a lot of bad artist names etc when doing that. I even tried to use lidarr api for forced import  but i am not good enough at scripting to get it to work.

**Reply by RandomNinjaAtk** [reply]
A script doesn't already exist to do what your looking for, I was just saying it wouldn't be difficult because the audio script essentially does this already for its own processes. However, you would need to create a custom script to do what your looking for specifically.

API call used to tell lidarr to import: https://github.com/RandomNinjaAtk/docker-lidarr-extended/blob/57a7e86bcd3957a98a5d64aebe04daed8864f4c9/root/scripts/Audio.sh#L869

You would just need to replace the variables for your custom script:
$lidarrUrl = Lidarr URL
${lidarrApiKey} = Lidarr API Key
$1 = /Folder/to/import (single album folder (not multiple albums); must be accessible by lidarr)
